### PR TITLE
Decouple verify client logic from phone auth

### PR DESCRIPTION
**Pull request by renkelvin**

### CODE REVIEW

File: Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m [hunk 1]
```diff
@@ -377,12 +355,49 @@ - (void)verifyClientWithCompletion:(FIRVerifyClientCallback)completion {
...
+                                            eventID:eventID
+                                           authType:kAuthTypeVerifyApp
+                                     callbackScheme:self->_callbackScheme];
+    };
+    [self->_auth.authURLPresenter presentURL:reCAPTCHAURL
+                                  UIDelegate:UIDelegate
+                             callbackMatcher:callbackMatcher
+                                  completion:^(NSURL *_Nullable callbackURL,
+                                               NSError *_Nullable error) {
+                                    if (error) {
```
Comment (Yue-Wang-Google) [inline Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m]: indent

File: Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m [hunk 2]
```diff
@@ -346,15 +314,17 @@ - (void)verifyClientAndSendVerificationCodeToPhoneNumber:(NSString *)phoneNumber
...
   if (_auth.appCredentialManager.credential) {
-    completion(_auth.appCredentialManager.credential, nil);
+    completion(_auth.appCredentialManager.credential, nil, nil);
     return;
   }
   [_auth.tokenManager getTokenWithCallback:^(FIRAuthAPNSToken *_Nullable token,
                                              NSError *_Nullable error) {
     if (!token) {
-      completion(nil, [FIRAuthErrorUtils missingAppTokenErrorWithUnderlyingError:error]);
+//      completion(nil, nil, [FIRAuthErrorUtils missingAppTokenErrorWithUnderlyingError:error]);
```
Comment (Yue-Wang-Google) [inline Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m]: ?

File: Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m [hunk 3]
```diff
@@ -306,18 +263,28 @@ - (void)verifyClientAndSendVerificationCodeToPhoneNumber:(NSString *)phoneNumber
...
+    FIRSendVerificationCodeRequest *request;
+    if (appCredential) {
+      request =
+      [[FIRSendVerificationCodeRequest alloc] initWithPhoneNumber:phoneNumber
+                                                    appCredential:appCredential
+                                                   reCAPTCHAToken:nil
+                                             requestConfiguration:self->_auth.requestConfiguration];
+    } else if (reCAPTCHAToken) {
+      request =
+      [[FIRSendVerificationCodeRequest alloc] initWithPhoneNumber:phoneNumber
```
Comment (Yue-Wang-Google) [inline Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m]: 4 sp indent

File: Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m [hunk 4]
```diff
@@ -306,18 +263,28 @@ - (void)verifyClientAndSendVerificationCodeToPhoneNumber:(NSString *)phoneNumber
...
     }
-    FIRSendVerificationCodeRequest *request =
-        [[FIRSendVerificationCodeRequest alloc] initWithPhoneNumber:phoneNumber
-                                                     appCredential:appCredential
-                                                    reCAPTCHAToken:nil
-                                              requestConfiguration:
-                                                  self->_auth.requestConfiguration];
+    FIRSendVerificationCodeRequest *request;
+    if (appCredential) {
+      request =
```
Comment (Yue-Wang-Google) [inline Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m]: 4 sp indent

File: Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m (hunk 2 above)
Comment (renkelvin) [reply Firebase/Auth/Source/AuthProvider/Phone/FIRPhoneAuthProvider.m]: Forget to remove.

**Pull request merged**